Fix support for Tower 3.7

matburt opened the pull request:
Work-in-progress towards supporting Tower 3.7 

Closes #44.

matburt committed:
Work towards supporting 3.7

matburt commented on deploy/tower-operator.yaml:
```diff
@@ -113,14 +113,14 @@ spec:
             - /tmp/ansible-operator/runner
             - stdout
-          image: "geerlingguy/tower-operator:0.3.0"
+          image: "matburt/tower-operator:latest"
```
I totally did this to make it easier on me, I'll circle back to this later.

matburt committed:
More configuration

matburt committed:
Need to be specific on the container we call the commands on

shanemcd commented on roles/tower/tasks/initialize.yml:
```diff
@@ -12,7 +12,7 @@
 - name: Create Tower super user via Django if it doesn't exist.
   shell: >-
-    kubectl exec -n {{ meta.namespace }} {{ tower_pod_name }} -- bash -c
+    kubectl exec -n {{ meta.namespace }} -c tower-task {{ tower_pod_name }} -- bash -c
```
I _think_ this will be `awx-task` on upstream deployments.

geerlingguy replied:
If that's the case (and in CI, we test with `awx` since it's difficult to automate the licensing?), should the container name be a variable?

shanemcd replied:
Yeah, I think we can use `meta.name` here too.

matburt replied:
I don't think `meta.name` will work here. This is the actual container that we need to run again. `meta.name` will just be `tower`

matburt commented:
Getting a lot closer now:

`redis.exceptions.ConnectionError: Error 110 connecting to tower-redis.ansible-tower.svc.cluster.local:6379. Connection timed out.`

`RuntimeError: No instance found with the current cluster host id`

`
django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: libpq.so.rh-postgresql10-5: cannot open shared object file: No such file or directory
unable to load app 0 (mountpoint='') (callable not found or import error)
`

all from the task container


This is from the web container

`
2020-05-29 23:59:23,772 INFO exited: awx-rsyslogd (exit status 1; not expected)
rsyslogd: cannot create '/var/run/awx-rsyslog/rsyslog.sock': No such file or directory [v8.1911.0-7.el7at try https://www.rsyslog.com/e/2176 ]
rsyslogd: imuxsock does not run because we could not aquire any socket  [v8.1911.0-7.el7at]
rsyslogd: activation of module imuxsock failed [v8.1911.0-7.el7at]
rsyslogd: error writing pid file (creation stage)
: No such file or directory
rsyslogd: run failed with error -3000 (see rsyslog.h or try https://www.rsyslog.com/e/3000 to learn what that number means)
`

shanemcd committed:
Get 3.7 booting and running jobs

There's still some polishing that needs to happen, but this gets things running

shanemcd committed:
Move sensitive info out of environment, get websocket backplane working

matburt committed:
Merge remote-tracking branch 'shanemcd/37_support' into 37_support

matburt commented:
So close!

![image](https://user-images.githubusercontent.com/89544/83341669-15b1ac80-a2b4-11ea-97e3-b6efee0c2eda.png)

This leads me to believe that perhaps the values defined on the CRD aren't getting picked up and the role is just using whatever is in the role defaults.

shanemcd committed:
Fix external logging

matburt committed:
Merge remote-tracking branch 'shanemcd/37_support' into 37_support

matburt committed:
Be specific on the route api version,make sure we set a default tower broadcast secret

geerlingguy commented on roles/tower/templates/supervisor.yaml.j2:
```diff
@@ -0,0 +1,145 @@
+    app: tower
+data:
+  supervisor-web-config: |
+    [supervisord]
```
Isn't all this config in the image itself? Why does this need to be the responsibility of the operator? (When I'm inspecting the AWX images, at least, I see supervisord setting itself up correctly)

shanemcd replied:
This is something we're actively working on fixing. The problem is downstream only... In 3.7.0, we this moved out of the image built in Red Hat's infrastructure. This was a bad idea, and will be fixed in an upcoming 3.7.x release. But for now, we need this because we need to override this: https://github.com/geerlingguy/tower-operator/pull/45/files#diff-858c0f683bcf356ee3c28b5d41dd2daeR43

Note, this will break AWX... I need to fix this up before merging this.

geerlingguy commented:
Changed the title since technically #43 at least worked to spin things up, though not correctly for Tower itself. This PR is to get the operator building Tower more correctly for 3.7 (and in general).

I want to make sure we can get CI passing though, and that latest AWX can work correctly with any required settings differences configurable.

matburt committed:
Point at the ansible dockerhub

geerlingguy commented on deploy/tower-operator.yaml:
```diff
@@ -113,14 +113,14 @@ spec:
             - /tmp/ansible-operator/runner
             - stdout
-          image: "matburt/tower-operator:latest"
+          image: "ansible/tower-operator:latest"
```
(Just as a note, mostly for anyone else reading this in the immediate future, https://hub.docker.com/r/ansible/tower-operator doesn't yet exist.)

shanemcd committed:
Fix upstream install path

Note that this added the required attribute spec.deployment_type. This _can_ go
away after the supervisor configs move back into the downstream images, but the
pattern may prove useful for other things in the future.

matburt committed:
Merge pull request #3 from shanemcd/37_support

Fix upstream install path

matburt committed:
Fixing up lint errors

matburt committed:
Fix up Tower CRD variables for the proper deploy type

matburt committed:
Fixing up molecule

* Limiting resources so that things can start in travis
* Making a dedicated molecule test CR

matburt committed:
Bump version to v1beta1

geerlingguy commented on deploy/tower-operator.yaml:
```diff
@@ -113,14 +113,14 @@ spec:
             - /tmp/ansible-operator/runner
             - stdout
-          image: "geerlingguy/tower-operator:0.3.0"
+          image: "ansible/tower-operator:latest"
```
@matburt - Can you push up a version `0.4.0` tag, that way if someone installs the operator today and the underlying image at `:latest` is different in a month or two, it won't break from under them.

geerlingguy replied:
Nevermind, there is now a `0.4.0` tag :)

matburt committed:
Set operator version to 0.4.0

geerlingguy commented on molecule/test-minikube/converge.yml:
```diff
@@ -73,7 +73,7 @@
             api_version: v1
 
-        - name: Create the tower.ansible.com/v1alpha1.Tower
+        - name: Create the tower.ansible.com Tower Resource
```
For consistency, we should have this line and the one in `test-local/converge.yml` be the same. It was using the CRD-based-path, but we can just call it 'Create the Tower Custom Resource' for more readability and to take the version out of this string (easier maintenance).

geerlingguy commented on roles/tower/templates/launch_awx.yaml.j2:
```diff
@@ -0,0 +1,60 @@
+    ANSIBLE_REMOTE_TEMP=/tmp ANSIBLE_LOCAL_TEMP=/tmp ansible -i "127.0.0.1," -c local -v -m wait_for -a "host=$DATABASE_HOST port=$DATABASE_PORT" all
+    ANSIBLE_REMOTE_TEMP=/tmp ANSIBLE_LOCAL_TEMP=/tmp ansible -i "127.0.0.1," -c local -v -m wait_for -a "path=/var/run/redis/redis.sock" all
+
+
```
I'm not sure if this is a particular style decision, but I typically keep one newline spacing in shell scripts. Unless there's a good reason not to, can you remove the extra newline here?

matburt committed:
Fix up some naming and spacing issues
